vertical-pod-autoscaler/hack/vpa-process-yaml.sh

```diff
@@ -18,14 +18,27 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
-
 function print_help {
   echo "ERROR! Usage: vpa-process-yaml.sh <YAML files>+"
   echo "Script will output content of YAML files separated with YAML document"
   echo "separator and substituting REGISTRY and TAG for pod images"
 }
 
+function apply_feature_gate() {
+  filename=$1
+  if [ -n "${FEATURE_GATES}" ]; then
+    if [[ "${filename}" == *"admission-controller"* ]]; then
+      kubectl patch --type=json --local -p='[{"op": "add", "path": "/spec/template/spec/containers/0/args/-", "value": "--feature-gates='"${FEATURE_GATES}"'"}]' -o yaml -f -
```

Comment:
I haven't tried running this locally yet, but I don't see how it works.
I assume this needs to be:
```suggestion
      kubectl patch --type=json --local -p='[{"op": "add", "path": "/spec/template/spec/containers/0/args/-", "value": "--feature-gates='"${FEATURE_GATES}"'"}]' -o yaml -f "${filename}" 
```

Reply:
The function itself is supposed to take input from stdin and output to stdout (like what we eventually do in `kubectl apply -f -` inside `deploy-for-e2e-locally.sh`.
```bash
sed -e "s,${DEFAULT_REGISTRY}/\([a-z-]*\):.*,${REGISTRY_TO_APPLY}/\1:${TAG_TO_APPLY}," $i | apply_feature_gate "$(basename "$i")"`
```
(See the piping from the `sed` command into that `apply_feature_gate` function.)

I just thought it was easier to do it that way instead of copying the output of the sed command to a tmp file or a variable and then applying the patch separately, but I'm happy to know if there are more elegant solutions.

I can also add a comment to clarify what the function does, if we want to document.

Reply:
Oh yes, that makes sense. I was focusing on the function, rather than how it was being called. The $filename was confusing me.
It makes sense, I think it's fine to leave as is.